pkgs/development/libraries/science/math/nccl/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchFromGitHub
+, gcc5, eject, cudatoolkit
+}:
+
+stdenv.mkDerivation rec {
+  name = "cudatoolkit-${cudatoolkit.majorVersion}-nccl-${version}";
+  version = "1.3.4-1";
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "nccl";
+    rev = "649f04d07711440ec2699ce7ec1af4ab71d6dbf9";
+    sha256 = "0fvnrfn572lc6i2a3xyhbifm53ivcrr46z6cqr3b0bwb1iq79m7q";
+  };
+
+  nativeBuildInputs = [
+    gcc5
+    eject
+  ];
+
+  propagatedBuildInputs = [
+    cudatoolkit
+  ];
+
+  BUILDDIR = "./";
+  PREFIX = "$out";
```

Comment:
This will set the `PREFIX` variable to literal `$out` string. The `out` variable needs to be interpolated somewhere, either by shell using `preInstall = ''export PREFIX="$out"'';`, or by make as suggested below. Correctly setting the variable will also allow you to get rid of the `postInstall` section.